### PR TITLE
Tech Page

### PR DESCRIPTION
**Pull request by lazydevorg**
### Motivation

- [Related story](https://trello.com/c/uBtPANmP/24-tech-page%3A-roundtable-slice)

Sets up the tech page, routing for the tech page and the general layout.

This allows us to implement the various slices required for MVP. The tech roundtable slice has been included in this PR

### TODO:

- [ ] Tech roundtable pdf
- [x] Update Tech Roundtable image to the new one, provided by Clem
- [x] Blog post selection for "Tried and tested" and "Growing Trends"
- [ ] Changing current links around the site to point to new tech page, and also on the old website (case study pages) - ready to go in this PR - https://github.com/redbadger/website-next-proxy/pull/35

### Test plan

- Responsiveness
- Link to roundtable pdf
- GA event firing on roundtable pdf link click

### Pre-merge checklist

- [ ] Documentation
- [ ] Unit tests
- [ ] Reviews
  - [ ] Code review
  - [ ] Design review
- [ ] Manual testing
  - [ ] Windows 7 IE 11
  - [ ] Windows 10 Edge
  - [ ] Mac OS El Capitan Safari
  - [ ] Mac OS El Capitan Chrome
  - [ ] Mac OS El Capitan Firefox
  - [ ] iOS 9 Safari
  - [ ] Android 6 Chrome
  - [ ] Listen to the site on a screen reader
  - [ ] Navigate the site using the keyboard only
- [ ] Tester approved

### Pre-deploy checklist
 - [ ] Merge the proxy redirect configuration https://github.com/redbadger/website-next-proxy/pull/35

### CODE REVIEW

File: site/compiler/people.test.js [hunk 1]
```diff
@@ -34,10 +34,10 @@ describe('site/compiler', () => {
         badger: { [a.slug]: a },
       }, createStateNavigator());
 
-      expect(routes.length).to.equal(14);
-      expect(routes[10].filePath).to.equal('about-us/people/index.html');
-      expect(routes[11].filePath).to.equal('about-us/people/category/engineering/index.html');
-      expect(routes[12].filePath).to.equal('about-us/people/category/leadership/index.html');
+      expect(routes.length).to.equal(15);
```
Comment (Xanndarr) [inline site/compiler/people.test.js]: -_-

These tests are a pain, but due to the way the router generates pages, there's not much we can do to fix this

**Comment by gregdorward** [comment]
@Xanndarr I'll add anything I find to the checklist below, so we can track them as they're reworked.

- [ ] Increase width of Tech Roundtable blurb

Design:
screen shot 2017-06-06 at 13 47 17

Actual:
screen shot 2017-06-06 at 13 47 01

File: site/pages/technology/style.css [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  composes: serif from "../../css/typography/_fonts.css";
+  color: badgerBlack;
+  line-height: 1.21;
+}
+
+.latestRoundTableLink {
+  font-family: 'ProximaNova-Bold', sans-serif;
+  border-bottom: 2px solid #E5051A;
+  text-decoration: none;
+  color: #fc1d42;
```
Comment (gregdorward) [inline site/pages/technology/style.css]: Can we use badgerRedOnWhite here please @Xanndarr? It should pass accessibility tests then.

**Comment by Xanndarr** [comment]
@gregdorward sorted all that out, ready for re-test

File: site/pages/technology/style.css [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+.description {
+  composes: fontS from "../../css/typography/_fonts.css";
+  composes: serif from "../../css/typography/_fonts.css";
+  color: badgerBlack;
+  line-height: 1.21;
+}
+
+.latestRoundTableLink {
+  font-family: 'ProximaNova-Bold', sans-serif;
+  border-bottom: 2px solid #E5051A;
```
Comment (Xanndarr) [inline site/pages/technology/style.css]: Missed that! Fixed now

**Comment by Xanndarr** [comment]
Fixed those left over comments from yesterday I missed!

File: site/pages/technology/style.css [hunk 4]
```diff
@@ -94,6 +94,11 @@
   height: 20px;
 }
 
+.arrow svg g {
```
Comment (gregdorward) [inline site/pages/technology/style.css]: Can we apply this colour to the style__latestRoundTableLink___2k8YE as well please?

**Comment by gregdorward** [comment]
One more review comment added for a small tweak, but other than that, this is looking good

**Comment by gregdorward** [comment]
Tested and passed, but I've added the 'Don't merge' label, as this needs to be bundled together with the other Tech Page MVP tickets and tested as a whole before being released

**Comment by chris-brookes** [comment]
@Xanndarr - In IE11 the round table image is not changing size to stay in view when you change the width of the browser. Part of the image is cut off when on the min width of each layout.

screen shot 2017-06-12 at 15 42 57

**Comment by Xanndarr** [comment]
@chris-brookes fixed in later PR

**Comment by chris-brookes** [comment]
@lazydevorg - The new pdf document for the tech round table is here: https://drive.google.com/drive/folders/0B0xlwGMZXlw8VDZBWWFGcjZxQ1E

This needs to be setup for download from the link in the intro section at the top of the page.

File: site/compiler/people.test.js (hunk 1 above)
Comment (romanschejbal) [inline site/compiler/people.test.js]: What about if we use jest's snapshot testing for these?

E.g.
```
expect(routes.length).to.equal(15);
expect(routes.map(route => route.filePath)).toMatchSnapshot();
```

File: site/fetchers/blog-posts/test.js [hunk 5]
```diff
@@ -30,7 +30,7 @@ describe('featured blog posts fetcher', () => {
   it('maps blog posts to expected data shape', () => {
     const data = mapDataToState(fixture());
     const post = data[0];
-    expect(post).to.have.all.keys('slug', 'category', 'title', 'author');
+    expect(post).to.have.all.keys('slug', 'category', 'title', 'author', 'date', 'excerpt');
```
Comment (romanschejbal) [inline site/fetchers/blog-posts/test.js]: Should we test the values as well? I'd suggest to use [toMatchObject](http://facebook.github.io/jest/docs/en/expect.html#tomatchobjectobject) here.

File: site/pages/technology/blog-slice/blog-entry/index.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+export type BlogPost = {
+  slug: string,
+  category: string,
+  title: string,
+  author: Author,
+  excerpt: string,
+  date: Date,
+};
+
+const BlogEntry = ({ blogPost, altStyle }: { blogPost: BlogPost, altStyle?: boolean }) => {
```
Comment (romanschejbal) [inline site/pages/technology/blog-slice/blog-entry/index.js]: This line feels a bit long, are we not using Prettier on this project?

File: site/compiler/people.test.js (hunk 1 above)
Comment (romanschejbal) [inline site/compiler/people.test.js]: Maybe, we can even ditch the length.. What do you think?

File: site/pages/technology/blog-slice/blog-entry/index.js (hunk 6 above)
Comment (Xanndarr) [inline site/pages/technology/blog-slice/blog-entry/index.js]: Prettier is happy with this

Comment (romanschejbal) [inline site/pages/technology/blog-slice/blog-entry/index.js]: 👍  cool 😊  

Everything looks longer on this small screen 😆

**Comment by gregdorward** [comment]
@romanschejbal, are these changes things that would prevent approval, or are they nice to haves? I only ask because we have to get this out today. We don't want to be reckless, but if there are improvements we can make to tests, i.e. incorporating snapshot testing, maybe we could do that in a follow-up ticket?

**Pull request merged**